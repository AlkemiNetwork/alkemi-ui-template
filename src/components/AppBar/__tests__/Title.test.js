import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import ButtonBase from "@material-ui/core/ButtonBase";

import AppBarTitle from "../Title";

describe("<AppBarTitle />", () => {
    let wrapper;
    let props = {};

    beforeEach(() => {
        props.onClick = jest.fn();
        wrapper = shallow(<AppBarTitle {...props} />);
    });

    it("matches snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders 1 button", () => {
        expect(wrapper.find(ButtonBase).length).toEqual(1);
    });

    it("button is functional", () => {
        wrapper
            .find(ButtonBase)
            .at(0)
            .simulate("click");
        expect(props.onClick.mock.calls.length).toEqual(1);
    });
});
