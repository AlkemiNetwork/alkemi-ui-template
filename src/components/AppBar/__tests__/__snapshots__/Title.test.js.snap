// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBarTitle /> matches snapshot 1`] = `
<WithStyles(ButtonBase)
  onClick={[MockFunction]}
>
  <WithStyles(Typography)
    color="inherit"
    noWrap={true}
    variant="title"
  >
    ConsenSys React Boilerplate
  </WithStyles(Typography)>
</WithStyles(ButtonBase)>
`;
